src/hooks/useAxios.ts

```diff
@@ -0,0 +1,41 @@
+import axios, { AxiosRequestConfig } from "axios";
+import { useEffect, useState } from "react";
+
+type Methods = "get" | "post" | "put" | "delete" | "patch";
+
+type UseAxiosResult = {
+  data: any;
+  loading: boolean;
+  error: Error | null;
+};
+
+const useAxios = (
+  url: string,
+  method: Methods = "get",
+  config?: AxiosRequestConfig
+): UseAxiosResult => {
+  const intialState: UseAxiosResult = {
+    data: null,
+    loading: true,
+    error: null,
+  };
+
+  const [state, setState] = useState(intialState);
+
+  useEffect(() => {
+    const request = axios[method](url, {

```

Comment:
how will the post, put, patch request work?
Will the data be sent in the config only or what?

Reply:
This way 
```useAxios(patchUrl, "patch", {
        data: {
          patch: ["members-site", "real-dev", "real-dev-squad", "patch"],
        },
        withCredentials: true,
      })
    );```